authorization-server/src/modules/auth/auth.controller.ts

```diff
@@ -189,24 +191,70 @@ export class AuthController {
   @UseGuards(ValidRefreshTokenGuard)
   @ApiBody({ type: RefreshDto })
   @ApiOkResponse({ type: LoginResponseDto })
-  async refresh(
+  // backwards compatibility
+  async refreshWithPost(
     @Body() body: RefreshDto,
     @Res({ passthrough: true }) res: Response,
   ): Promise<LoginResponseDto | undefined> {
-    const { accessToken, refreshToken } = await this.authService.refreshTokens(
-      body.refreshToken,
+    return this.refreshCommon(body.refreshToken, res);
+  }
+
+  @Get('refresh_token')
+  //compatibile with the SSI-HUB implementation
+  async refreshWithGet(
+    @Req() req: Request,
+    @Res({ passthrough: true }) res: Response,
+    @Query('refresh_token') refreshTokenFromQueryString: string,
+  ): Promise<LoginResponseDto | undefined> {
+    let refreshToken: string;
+
+    if (this.configService.get<boolean>('AUTH_COOKIE_ENABLED')) {
+      refreshToken =
+        refreshTokenFromQueryString ||
+        (req.cookies || {})[
+          this.configService.get('AUTH_COOKIE_NAME_REFRESH_TOKEN')
+        ];
+    } else {
+      refreshToken = refreshTokenFromQueryString;
+    }
+
+    if (!refreshToken) {
+      this.logger.warn(`no refresh token provided`);
+      throw new UnauthorizedException();
+    }
+
+    const refreshTokenIsValid = await this.authService.validateRefreshToken(
+      refreshToken,
     );
 
+    if (!refreshTokenIsValid) {
+      this.logger.warn(`invalid refresh token provided`);
+      throw new ForbiddenException();
+    }
+
+    return await this.refreshCommon(refreshToken, res);
+  }
+
+  async refreshCommon(
```

Comment:
Small thing but making it private can make it a bit clearly to readers that it is internal to class
```suggestion
  private async refreshCommon(
```

Reply:
I agree however, writing tests for private methods becomes more ugly.